src/System.Composition.AttributedModel/src/System/Composition/PartMetadataAttribute.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace System.Composition
+{
+    /// <summary>
+    ///     Specifies metadata for a type to be used as a part.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public class PartMetadataAttribute : Attribute
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="PartMetadataAttribute"/> with the 
+        ///     specified name and metadata value.
+        /// </summary>
+        /// <param name="name">
+        ///     A <see cref="String"/> containing the name of the metadata value; or 
+        ///     <see langword="null"/> to use an empty string ("").
+        /// </param>
+        /// <param name="value">
+        ///     An <see cref="object"/> containing the metadata value. This can be 
+        ///     <see langword="null"/>.
+        /// </param>
+        public PartMetadataAttribute(string name, object value)
+        {
+            if (name == null) throw new ArgumentNullException("name");
+
+            Name = name;
+            Value = value;
+        }
+
+        /// <summary>
+        ///     Gets the name of the metadata value.
+        /// </summary>
+        /// <value>
+        ///     A <see cref="String"/> containing the name of the metadata value.
+        /// </value>
+        public string Name
+        {
+            get;
+            private set;
```

Comment:
In some of these attributes, private readonly fields are being used as the backing store, whereas in others private setters are being used on the properties.  Is there a rhyme/reason to when each approach is used?